aws/rust-runtime/aws-types/src/credentials/provider.rs

```diff
@@ -4,147 +4,173 @@
-                write!(f, "The credential provider was not enabled: {}", context)
-            }
-            CredentialsError::ProviderTimedOut(d) => write!(
-                f,
-                "Credentials provider timed out after {} seconds",
-                d.as_secs()
-            ),
-            CredentialsError::Unhandled { cause } => {
-                write!(f, "Unexpected credentials error: {}", cause)
-            }
-            CredentialsError::InvalidConfiguration { cause } => {
-                write!(
+    impl fmt::Display for CredentialsError {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            match self {
+                CredentialsError::CredentialsNotLoaded(_) => {
+                    write!(f, "the credential provider was not enabled")
+                }
+                CredentialsError::ProviderTimedOut(details) => write!(
                     f,
-                    "The credentials provider was not properly configured: {}",
-                    cause
-                )
-            }
-            CredentialsError::ProviderError { cause } => {
-                write!(f, "An error occurred while loading credentials: {}", cause)
+                    "credentials provider timed out after {} seconds",
+                    details.timeout_duration.as_secs()
+                ),
+                CredentialsError::InvalidConfiguration(_) => {
+                    write!(f, "the credentials provider was not properly configured")
+                }
+                CredentialsError::ProviderError(_) => {
+                    write!(f, "an error occurred while loading credentials")
+                }
+                CredentialsError::Unhandled(_) => {
+                    write!(f, "unexpected credentials error")
+                }
             }
         }
     }
-}
 
-impl Error for CredentialsError {
-    fn source(&self) -> Option<&(dyn Error + 'static)> {
-        match self {
-            CredentialsError::Unhandled { cause }
-            | CredentialsError::ProviderError { cause }
-            | CredentialsError::InvalidConfiguration { cause } => Some(cause.as_ref() as _),
-            CredentialsError::CredentialsNotLoaded { context } => Some(context.as_ref() as _),
-            _ => None,
+    impl Error for CredentialsError {
+        fn source(&self) -> Option<&(dyn Error + 'static)> {
+            match self {
+                CredentialsError::CredentialsNotLoaded(details) => {
+                    Some(details.source.as_ref() as _)
+                }
+                CredentialsError::ProviderTimedOut(_) => None,
+                CredentialsError::InvalidConfiguration(details) => {
+                    Some(details.source.as_ref() as _)
+                }
+                CredentialsError::ProviderError(details) => Some(details.source.as_ref() as _),
+                CredentialsError::Unhandled(details) => Some(details.source.as_ref() as _),
+            }
```

Comment:
The cast isn't actually needed here I think.
```suggestion
    impl Error for CredentialsError {
        fn source(&self) -> Option<&(dyn Error + 'static)> {
            match self {
                CredentialsError::CredentialsNotLoaded(details) => Some(details.source.as_ref()),
                CredentialsError::ProviderTimedOut(_) => None,
                CredentialsError::InvalidConfiguration(details) => Some(details.source.as_ref()),
                CredentialsError::ProviderError(details) => Some(details.source.as_ref()),
                CredentialsError::Unhandled(details) => Some(details.source.as_ref()),
            }
```